### PR TITLE
Backslash on Unix is a valid path character- need to fix up separators b...

### PR DESCRIPTION
**Pull request by JeremyKuhne**
Backslash on Unix is a valid path character- need to fix up separators before invoking.

### CODE REVIEW

**Comment by JeremyKuhne** [comment]
@david-mitchell, @akoeplinger: Here is a simpler solution for the nuget restore issue on Unix I believe. I've been doing quite a bit of thinking about alternatives- I think a targeted solution is best.

**Comment by akoeplinger** [comment]
I'm fine with it as a temporary workaround until we get a built-in MSBuild property function (note that this still won't work for xbuild as inline tasks don't work there at all).

**Comment by JeremyKuhne** [comment]
@akoeplinger any suggestions on how to address the bootstrap download for xbuild without checking in binaries?

**Comment by akoeplinger** [comment]
Actually, after having thought about it more, the only thing that is an issue is the backslash in ` $(SourceDir).nuget\packages.config`. We could just flip that to a normal slash and everything would work. This is also what MSBuild is doing in their xplat branch: [dir.targets#L60](https://github.com/Microsoft/msbuild/blob/6a8d6e20bcfee1e0f1114dfd3a8459d1c3ff87a3/dir.targets#L60)

**Comment by JeremyKuhne** [comment]
I assume the `$(SourceDir)` would need flipping too.

File: dir.props [hunk 1]
```diff
@@ -15,10 +15,12 @@
     <ObjDir Condition="'$(ObjDir)'==''">$(BinDir)obj\</ObjDir>
     <TestWorkingDir Condition="'$(TestWorkingDir)'==''">$(BinDir)tests\</TestWorkingDir>
     <PackagesOutDir Condition="'$(PackagesOutDir)'==''">$(BinDir)packages\</PackagesOutDir>
-    
+
     <!-- Input Directories -->
     <PackagesDir Condition="'$(PackagesDir)'==''">$(ProjectDir)packages\</PackagesDir>
     <ToolsDir Condition="'$(ToolsDir)'==''">$(PackagesDir)Microsoft.DotNet.BuildTools.$(BuildToolsVersion)\lib\</ToolsDir>
+
+    <DirectorySeparator Condition="'$(DirectorySeparator)'==''">$([System.IO.Path]::DirectorySeparatorChar)</DirectorySeparator>
```
Comment (weshaggard) [inline dir.props]: We set this but don't use it anywhere,

Comment (JeremyKuhne) [inline dir.props]: I was using it in the other file- figured it was good to be set globally so others could do the same thing. That said, it was pointed out that the only issue was the config file. As such I'm going even simpler here.

**Comment by stephentoub** [comment]
LGTM

**Comment by weshaggard** [comment]
I don't think the current change is enough. If you look at the MSBuild repro (https://github.com/Microsoft/msbuild/blob/6a8d6e20bcfee1e0f1114dfd3a8459d1c3ff87a3/dir.props#L9) they also change the separator for all the other paths in dir.props, which I expect we would also need.

**Comment by JeremyKuhne** [comment]
I'll update later this week once I hopefully get everything building myself.

Sent from my Windows Phone
________________________________

**Pull request closed without merging**